subprojects/logging/src/main/java/org/gradle/internal/logging/console/ThrottlingOutputEventListener.java

```diff
@@ -26,27 +26,38 @@
 import java.util.List;
 import java.util.concurrent.Executors;
 import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
 
 /**
  * Queue output events to be forwarded and schedule flush when time passed or if end of build is signalled.
  */
 public class ThrottlingOutputEventListener implements OutputEventListener {
-    private final BatchOutputEventListener listener;
+    private final OutputEventListener listener;
 
     private final ScheduledExecutorService executor;
     private final TimeProvider timeProvider;
     private final int throttleMs;
     private final Object lock = new Object();
 
+    private final Runnable scheduledRenderNow = new Runnable() {
```

Comment:
The work was laid out to make the WIP renderer change easier. It also makes the rendering of the console more uniform as well as we can have a single point that decides when the console will be rerendered. We could change it if you feel there is a better way.